ibazel/profiler/profiler.js

```diff
@@ -0,0 +1,97 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+(function(){
+  function extractUrl() {
+    var url;
+    [...window.document.getElementsByTagName('script')].forEach(function (element) {
+      if (!url && element.src && element.src.match(/^[^:]+:\/\/(.*)\/z?profiler\.js(?:\?(.*))?$/)) {
+        const parsed = document.createElement('a');
+        parsed.href = element.src;
+        url = `${parsed.protocol}//${parsed.host}/profiler-event`;
+      }
+    });
+    return url;
```

Comment:
I think the return value to functions passed into `Array.prototype.forEach` is never used so you can save these bytes.

Reply:
This is the return value of `function extractUrl()`

Reply:
I'll do it the stylistic way with map, filter and reduce instead. Good suggestion